### ISSUE TITLE
Layer focus trap

### ISSUE THREAD

**Issue by halocline** [grommet/hpe-design-system#3841]
### Link to problem

https://design-system.hpe.design/components/layer

### Current problem

When a layer of type modal is opened, the focus should be trapped within that layer. Currently, when tabbing through the layer the focus within the Grommet app is trapped, but continued tabbing allows focus to go to other browser tabs, global header, etc.

### Expected behavior or solution

- [ ] Please review WCAG guidelines to verify whether focus can 'leak' to elements outside of application or should be trapped and only allowed to focus on elements within the layer.
- [ ] Check if an existing issue exists in Grommet
- [ ] Add "trap focus" guidance for modals within [Layer's accessibility section](https://design-system.hpe.design/components/layer#accessibility).

**Comment by MarinaCastejonHPE** [comment]
Thanks @halocline. Among all the WCAG documentation in their site, I found [this direct reference regarding the focus behaviour in modals](https://www.w3.org/WAI/WCAG22/Understanding/no-keyboard-trap.html) (see third example).

**Comment by jcfilben** [comment]
The `trapFocus` prop can be passed to Layer but we don't have that documented on Grommet site

**Comment by jcfilben** [comment]
Should be resolved by https://github.com/grommet/grommet/pull/7324
